jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -100,17 +99,16 @@ public class HttpConnection extends AbstractMetaDataConnection implements Runnab
     private final Lazy _attributes = new Lazy();
     private final DemandContentCallback _demandContentCallback = new DemandContentCallback();
     private final SendCallback _sendCallback = new SendCallback();
-    private final LongAdder bytesIn = new LongAdder();
-    private final LongAdder bytesOut = new LongAdder();
     private final AtomicBoolean _handling = new AtomicBoolean(false);
     private final HttpFields.Mutable _headerBuilder = HttpFields.build();
     private volatile RetainableByteBuffer _requestBuffer;
     private HttpFields.Mutable _trailers;
     private Runnable _onRequest;
+    private boolean _delayedForContent;
     private long _requests;
-    // TODO why is this not on HttpConfiguration?
-    private boolean _useInputDirectByteBuffers;
-    private boolean _useOutputDirectByteBuffers;
+    private long _responses;
+    private long _bytesIn;
+    private long _bytesOut;
```

Comment:
These were atomic to allow JMX to read them properly (through e.g. `ConnectionStatistics`) and avoid making them `volatile`.

Reply:
Previously it was a LongAdder, that is unnecessary as updates are not contended by multiple threads.   AtomicLong would be sufficient... but then there are infrequently read stats and will only ever be a snapshot, so I'm not even sure it is worthwhile using AtomicLong.  Perhaps we just put some memory barrier in the getters?  or just make them volatile (like _requestBuffer)